benchmark/compare.js

```diff
@@ -1,181 +1,73 @@
 'use strict';
-var usage = 'node benchmark/compare.js ' +
-            '<node-binary1> <node-binary2> ' +
-            '[--html] [--red|-r] [--green|-g] ' +
-            '[-- <type> [testFilter]]';
 
-var show = 'both';
-var nodes = [];
-var html = false;
-var benchmarks;
-
-for (var i = 2; i < process.argv.length; i++) {
-  var arg = process.argv[i];
-  switch (arg) {
-    case '--red': case '-r':
-      show = show === 'green' ? 'both' : 'red';
-      break;
-    case '--green': case '-g':
-      show = show === 'red' ? 'both' : 'green';
-      break;
-    case '--html':
-      html = true;
-      break;
-    case '-h': case '-?': case '--help':
-      console.log(usage);
-      process.exit(0);
-      break;
-    case '--':
-      benchmarks = [];
-      break;
-    default:
-      if (Array.isArray(benchmarks))
-        benchmarks.push(arg);
-      else
-        nodes.push(arg);
-      break;
-  }
+const fork = require('child_process').fork;
+const path = require('path');
+const CLI = require('./_cli.js');
+
+//
+// Parse arguments
+//
+const cli = CLI(`usage: ./node benchmark/compare.js <type> ...
```

Comment:
> I will try and change the argument order, this appears to cause a lot of confusion for many people, but I would love to understand why.

If I had to guess, I’d say it’s because that’s the order usually suggested in man pages and `--help` texts, and maybe because the positional arguments are the ones one is most likely to spend more time editing before hitting enter… idk, maybe there’s more to it.